nnstreamer/tensor_filter/tensor_filter_nntrainer.cc

```diff
@@ -258,36 +230,32 @@ static void nntrainer_close(const GstTensorFilterProperties *prop,
 
 static int nntrainer_loadModelFile(const GstTensorFilterProperties *prop,
                                    void **private_data) {
-  NNTrainer *nntrainer;
-  const gchar *model_file;
   if (prop->num_models != 1)
     return -1;
 
-  nntrainer = static_cast<NNTrainer *>(*private_data);
-  model_file = prop->model_files[0];
-
+  NNTrainer *nntrainer = static_cast<NNTrainer *>(*private_data);
+  const gchar *model_file = prop->model_files[0];
   if (nntrainer != NULL) {
     if (g_strcmp0(model_file, nntrainer->getModelConfig()) == 0)
       return 1; /* skipped */
 
     nntrainer_close(prop, private_data);
   }
 
-  nntrainer = new NNTrainer(model_file);
-  if (nntrainer == NULL) {
-    g_printerr("Failed to allocate memory for filter subplugin: nntrainer\n");
+  try {
+    nntrainer = new NNTrainer(model_file, prop);
+  } catch (std::bad_alloc &e) {
```

Comment:
I think this is redundant? `std::exception` can catch this exception too and do the same action.
Return value would be different but -1 is used in other places in this function as well.

Reply:
Done